### ISSUE TITLE
Swift Message hides automatically

### ISSUE THREAD

**Issue by iamsanketray123** [SwiftKickMobile/SwiftMessages#206]
I am using swift message in a `web view` during some transaction process. When the transaction begins, I display an Under progress, swift message. If it takes too long, I hide the previous one and display another. But sometimes, the first message hides automatically after a few seconds, even though the config.duration is set to `.forever`.  Running the app on device, iPhone 6. Latest Xcode and SwiftMessages version.

**Comment by wtmoose** [comment]
I don't see how that's possible. Can you share the code that displays the message?

**Comment by iamsanketray123** [comment]
https://paste.ofcode.org/g4RBimZeKPF422NFfWtZs5
Please check this. @wtmoose . I have a doubt that, it may be due the estimated progress thing of web view. Please check and lemme know.

**Comment by iamsanketray123** [comment]
@wtmoose I tried running the same code twice today. did not face any similar errors though. Its weird. Anyways do check the code. Maybe you will find something, that I missed.

**Comment by wtmoose** [comment]
I don’t have time to study a 400 line view controller. Could you post the code that displays the message (and nothing else)?

**Comment by wtmoose** [comment]
Also, as you say you’re using `.forever` duration and the message is hiding sooner than you expect. I’d say there’s a 99.9999% chance you’re calling `hide()` and you don’t know it. Have you tried putting a breakpoint on the line of code where you hide the message?

**Comment by iamsanketray123** [comment]
`                displayInitialMessage() //only enabled for QP
                timer = Timer.scheduledTimer(timeInterval: 1.0, target: self, selector: #selector(updateCountdown(timer:)), userInfo: nil, repeats: true)`

`
    var counter = 0 {
        didSet {
            if counter == 10 {
                print("Reached 10")
                display10SecondsWarning()
            }
            if counter == 20 {
                display20SecondsWarning()
                print("Reached 20")
            }
            if counter == 60 {
                print("Opening freshchat now")
                SwiftMessages.hide()
                SVProgressHUD.dismiss()
                sendErrorMessageToFreshchat()
            }
        }
    }
`

`    @objc func updateCountdown(timer: Timer) {
        counter += 1
        print(counter)
        
        if counter == 60 {
            timer.invalidate()
        }
    }`

`    func displayInitialMessage() {
        
//        SwiftMessages.hide()
        
        let view = MessageView.viewFromNib(layout: .cardView)
        view.configureTheme(.success)
        view.configureDropShadow()
        view.configureContent(title: nil, body: "Sending investment details to Fund House", iconImage: #imageLiteral(resourceName: "successIcon").withRenderingMode(.alwaysTemplate), iconText: nil, buttonImage: nil, buttonTitle: nil) { (_) in }
        view.button?.isHidden = true
        var config = SwiftMessages.Config()
        config.presentationStyle = .bottom
        config.duration = .seconds(seconds: 9)
        config.interactiveHide = false
        config.dimMode = .gray(interactive: true)
        config.preferredStatusBarStyle = .lightContent
        SwiftMessages.show(config: config, view: view)
        
    }
    
    func display10SecondsWarning() {
        
//        SwiftMessages.hide()
        
        let view = MessageView.viewFromNib(layout: .cardView)
        view.configureTheme(backgroundColor: UIColor.orange, foregroundColor: .white, iconImage: #imageLiteral(resourceName: "infoIcon.png").withRenderingMode(.alwaysTemplate), iconText: nil)
        view.configureDropShadow()
        
        view.configureContent(title: nil, body: "This transaction is taking longer than expected", iconImage: #imageLiteral(resourceName: "infoIcon").withRenderingMode(.alwaysTemplate), iconText: nil, buttonImage: nil, buttonTitle: nil) { (_) in }
        view.button?.isHidden = true
        var config = SwiftMessages.Config()
        config.presentationStyle = .bottom
        config.duration = .seconds(seconds: 9)
        
        config.interactiveHide = false
        config.dimMode = .gray(interactive: true)
        config.presentationContext = SwiftMessages.PresentationContext.window(windowLevel: UIWindowLevelStatusBar)
        
        SwiftMessages.show(config: config, view: view)
        
    }
    
    func display20SecondsWarning() {
        
//        SwiftMessages.hide()
        
        let view = MessageView.viewFromNib(layout: .cardView)
        view.configureTheme(.error)
        view.configureDropShadow()
        view.configureContent(title: "Transaction Pending", body: "You may go back to retry or contact us", iconImage: #imageLiteral(resourceName: "errorIcon.png").withRenderingMode(.alwaysTemplate), iconText: nil, buttonImage: nil, buttonTitle: "Contact Us") { (_) in
            
            SVProgressHUD.dismiss()
            SwiftMessages.hide()
            self.webView.stopLoading()
            self.sendErrorMessageToFreshchat()
            Freshchat.sharedInstance().showConversations(self)
            self.timer.invalidate()
            
//            SVProgressHUD.isVisible()
            
            self.messageSentToFreshchat = true
        }
        
        var config = SwiftMessages.Config()
        config.presentationStyle = .bottom
        config.duration = .forever
        config.interactiveHide = false
        config.dimMode = .gray(interactive: true)
        config.preferredStatusBarStyle = .lightContent
        config.presentationContext = SwiftMessages.PresentationContext.window(windowLevel: UIWindowLevelStatusBar)
        SwiftMessages.show(config: config, view: view)
        
        enableCancellation()
        
    }`

**Comment by iamsanketray123** [comment]
If I was calling hide, the swift messages would not work as expected like 8/10 times...it fails 2/10 times. anyways, I am copy pasting the code for you...

`displayInitialMessage() //only enabled for QP
                timer = Timer.scheduledTimer(timeInterval: 1.0, target: self, selector: #selector(updateCountdown(timer:)), userInfo: nil, repeats: true)
                `

`func displayInitialMessage() {
        

        
        let view = MessageView.viewFromNib(layout: .cardView)
        view.configureTheme(.success)
        view.configureDropShadow()
        view.configureContent(title: nil, body: "Sending investment details to Fund House", iconImage: #imageLiteral(resourceName: "successIcon").withRenderingMode(.alwaysTemplate), iconText: nil, buttonImage: nil, buttonTitle: nil) { (_) in }
        view.button?.isHidden = true
        var config = SwiftMessages.Config()
        config.presentationStyle = .bottom
        config.duration = .seconds(seconds: 9)
        config.interactiveHide = false
        config.dimMode = .gray(interactive: true)
        config.preferredStatusBarStyle = .lightContent
        SwiftMessages.show(config: config, view: view)
        
    }
    
    func display10SecondsWarning() {
        

        
        let view = MessageView.viewFromNib(layout: .cardView)
        view.configureTheme(backgroundColor: UIColor.orange, foregroundColor: .white, iconImage: #imageLiteral(resourceName: "infoIcon.png").withRenderingMode(.alwaysTemplate), iconText: nil)
        view.configureDropShadow()
        
        view.configureContent(title: nil, body: "This transaction is taking longer than expected", iconImage: #imageLiteral(resourceName: "infoIcon").withRenderingMode(.alwaysTemplate), iconText: nil, buttonImage: nil, buttonTitle: nil) { (_) in }
        view.button?.isHidden = true
        var config = SwiftMessages.Config()
        config.presentationStyle = .bottom
        config.duration = .seconds(seconds: 9)
        
        config.interactiveHide = false
        config.dimMode = .gray(interactive: true)
        config.presentationContext = SwiftMessages.PresentationContext.window(windowLevel: UIWindowLevelStatusBar)
        
        SwiftMessages.show(config: config, view: view)
        
    }
    
    func display20SecondsWarning() {
        

        
        let view = MessageView.viewFromNib(layout: .cardView)
        view.configureTheme(.error)
        view.configureDropShadow()
        view.configureContent(title: "Transaction Pending", body: "You may go back to retry or contact us", iconImage: #imageLiteral(resourceName: "errorIcon.png").withRenderingMode(.alwaysTemplate), iconText: nil, buttonImage: nil, buttonTitle: "Contact Us") { (_) in
            
            SVProgressHUD.dismiss()
            SwiftMessages.hide()
            self.webView.stopLoading()
            self.sendErrorMessageToFreshchat()
            Freshchat.sharedInstance().showConversations(self)
            self.timer.invalidate()
            

            
            self.messageSentToFreshchat = true
        }
        
        var config = SwiftMessages.Config()
        config.presentationStyle = .bottom
        config.duration = .forever
        config.interactiveHide = false
        config.dimMode = .gray(interactive: true)
        config.preferredStatusBarStyle = .lightContent
        config.presentationContext = SwiftMessages.PresentationContext.window(windowLevel: UIWindowLevelStatusBar)
        SwiftMessages.show(config: config, view: view)
        
        enableCancellation()
        
    }
    `

**Comment by iamsanketray123** [comment]
I switched from .forever to 9 seconds..but did face the same issue.

**Comment by wtmoose** [comment]
Put a breakpoint in `SwiftMessages.hideCurrent()` and you'll find out why it's being hidden.

**Comment by iamsanketray123** [comment]
@wtmoose I actually found the cause...why the message was auto hiding....i have a notification that runs continuously which detects the presence of Internet. Everytime the notification is triggered, SwiftMessages hides itself. Please fix!

**Comment by wtmoose** [comment]
I'm confused about what you want me to fix. Your code is calling `hide` when it shouldn't, correct?

**Comment by iamsanketray123** [comment]
@wtmoose yeah actually you are right, that was an issue from my end. Just one last question. Is it possible to name a SwiftMessage, so that I can detect, if that specific message shown and handle

**Comment by wtmoose** [comment]
Yep. There are APIs to query and hide messages by `id`. If you check the `MessageView` class, you will see that it has a default implementation for ID, but you can override that with any `id` you choose.

**Comment by iamsanketray123** [comment]
@wtmoose Thanks...you may resolve this
